src/test/java/hw1/DataProviders.java

```diff
@@ -0,0 +1,218 @@
+package hw1;
+
+import org.testng.annotations.DataProvider;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class DataProviders {
+
+    @DataProvider(name = "DataForSum")
+    public static Object[][] getDataForSumOperation() {
+
+        // Test cases:
+        // zero/zero, zero/pos, pos/zero, pos/pos, neg/neg, neg/pos, pos/neg, neg/pos(equal),
+        // pos/neg(equal),
+        // pos.Overflow, neg.Overflow
+
+        List<Long> firstOp = Arrays.asList(0L, 0L, 254L, 1000L, -500L, 500L, -1L,
+                247L, Long.MAX_VALUE, -10L);
+        List<Long> secondOp = Arrays.asList(0L, 899L, 0L, 315L, 239L, -239L, 1L,
+                -247L, 10L, Long.MIN_VALUE);
+        List<Long> result = Arrays.asList(0L, 899L, 254L, 1315L, -261L, 261L, 0L,
+                0L, -9223372036854775799L, 9223372036854775798L);
```

Comment:
I Would ask you to make `{{a, b, res}, {a, b, res}}` param placement because finding corresponding indices in three-four arrays is quite difficult

Reply:
done